sequencer/src/auction.rs

```diff
@@ -0,0 +1,304 @@
+                &bincode::serialize(&self.bundle.as_slice()).unwrap(),
+            );
+        comm.finalize()
+    }
+}
+
+impl Default for BidTxBody {
+    fn default() -> Self {
+        let message = ";)";
+        let mut commitment = [0u8; 32];
+        commitment[..message.len()].copy_from_slice(message.as_bytes());
+        let (account, key) = FeeAccount::generated_from_seed_indexed([0; 32], 0);
+        let nsid = NamespaceId::from(999);
+        Self {
+            url: Url::from_str("htts://sequencer:3939/request_budle").unwrap(),
+            account,
+            public_key: SequencerKey,
+            gas_price: FeeAmount::default(),
+            bid_amount: FeeAmount::default(),
+            slot: Slot::default(),
+            bundle: vec![nsid],
+        }
+    }
+}
+impl Default for BidTx {
+    fn default() -> Self {
+        let body = BidTxBody::default();
+        let commitment = body.commit();
+        let (account, key) = FeeAccount::generated_from_seed_indexed([0; 32], 0);
+        let signature = FeeAccount::sign_builder_message(&key, &commitment.as_ref()).unwrap();
+        Self { signature, body }
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize, Hash)]
+pub enum ExecutionError {
+    InvalidSignature,
+    InvalidPhase,
+}
+
+// TODO consider moving common functionality to trait.
+impl BidTx {
+    /// Executes `BidTx`.
+    /// * verify signature
+    /// * charge fee
+    /// * store state
+    // The rational behind the `Err` is to provide not only what
+    // failed, but for which varient. The entire Tx is probably
+    // overkill, but we can narrow down how much we want to know about
+    // Failed Tx in the future. Maybe we just want its name.
+    pub fn execute(&self, state: &ValidatedState) -> Result<(), (ExecutionError, FullNetworkTx)> {
+        if get_phase() != AuctionPhaseKind::Bid {
+            return Err((
+                ExecutionError::InvalidPhase,
+                FullNetworkTx::Bid(self.clone()),
+            ));
+        }
+        self.verify()
+            .map_err(|e| (e, FullNetworkTx::Bid(self.clone())))?;
+
+        // self.charge().map_err;
+        store_in_marketplace_state();
+
+        // TODO what do we return in good result?
+        Ok(())
+    }
+    // fn charge(&self) {}
+    /// Verify signature
+    fn verify(&self) -> Result<(), ExecutionError> {
+        if !self
+            .body
+            .account
+            .validate_builder_signature(&self.signature, self.body.commit().as_ref())
+        {
+            return Err(ExecutionError::InvalidSignature);
+        };
+
+        Ok(())
+    }
+}
+
+pub fn get_phase() -> AuctionPhaseKind {
+    AuctionPhaseKind::Bid
+}
+
+fn store_in_marketplace_state() {
+    unimplemented!();
+}
+
+enum ExecutionResult<T, E, K> {
+    Ok(T),
+    Err(E, K),
+}
+
+// enum ExecutionResult {
+//     Bid(String),
+// }
+
+/// A solution to one auction of sequencing rights for namespaces
+struct AuctionResultTx {
```

Comment:
Is this a single result, so an HTTP call to the solver to get the auction results for this view would be `Vec<AuctionResultTx>`?

Reply:
Also, where would the builder URL be stored in this case?

Reply:
I'm resolving this as it was discussed in zulip that this is a mistake left over from pre-JIT version.